packages/babel-helper-builder-react-jsx/src/index.js

```diff
@@ -156,82 +156,26 @@ You can turn on the 'throwIfNamespace' flag to bypass this warning.`,
     return state.call || t.callExpression(state.callee, args);
   }
 
-  function pushProps(_props, objs) {
-    if (!_props.length) return _props;
-
-    objs.push(t.objectExpression(_props));
-    return [];
-  }
-
-  /**
-   * The logic for this is quite terse. It's because we need to
-   * support spread elements. We loop over all attributes,
-   * breaking on spreads, we then push a new object containing
-   * all prior attributes to an array for later processing.
-   */
-
   function buildOpeningElementAttributes(attribs, file) {
-    let _props = [];
-    const objs = [];
-
-    const { useSpread = false } = file.opts;
-    if (typeof useSpread !== "boolean") {
-      throw new Error(
-        "transform-react-jsx currently only accepts a boolean option for " +
-          "useSpread (defaults to false)",
+    if ("useSpread" in file.opts) {
+      console.warn(
+        "useSpread is always enabled in Babel 8, please remove it from the config.\n" +
```

Comment:
Can we move this warning to the top-level plugin function?

Reply:
I think it is okay to stay here otherwise we need duplicate these between plugin-react-jsx and preset-react. In Babel 9 we can even remove these checks.

Reply:
Oh I realize that in presets we have to pass through `useSpread` and `useBuiltins` for this warning to pop up. It's a little bit awkward. I will move to plugin/preset then.